gslib/commands/cat.py

```diff
@@ -131,6 +131,10 @@ def RunCommand(self):
           range_matcher = re.compile(
               '^(?P<start>[0-9]+)-(?P<end>[0-9]*)$|^(?P<endslice>-[0-9]+)$')
           range_match = range_matcher.match(request_range)
+          # This if statement enures the full object is returned
+          # instead of throwing a CommandException.
+          if request_range == '-':
```

Comment:
Can this `if` block go above the `range_matcher` logic? `range_matcher` doesn't really matter if we can return earlier.